include/seqan3/range/utility.hpp

```diff
@@ -0,0 +1,89 @@
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Define various helper functionality.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/range/concept.hpp>
+#include <seqan3/core/concept/core.hpp>
+
+#include <range/v3/algorithm/equal.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Implements a generic equality comparator.
+ * \tparam lhs_range_type The first input range to compare (seqan3::input_range_concept).
+ * \tparam rhs_range_type The second input range to compare (seqan3::input_range_concept).
+ *
+ * The two ranges to be compared must satisfy the seqan3::input_range_concept and
+ * the reference type of both must be weakly comparable (seqan3::weakly_equality_comparable_concept).
+ * The function then delegates to the range_v3::equal function for comparison.
+ * \sa http://ericniebler.github.io/range-v3/group__group-views.html#gafde1fc2968ed1c0a30146a7231bb64fc
+ */
+template<typename lhs_range_type, typename rhs_range_type>
+//!\cond
+    requires input_range_concept<lhs_range_type> &&
+             input_range_concept<rhs_range_type> &&
+             weakly_equality_comparable_concept<typename lhs_range_type::reference, typename rhs_range_type::reference>
+//!\endcond
+bool operator==(lhs_range_type const & lhs, rhs_range_type const & rhs)
```

Comment:
we should define this consteexpr.
And if range::equal is not constexpr we need to define our own in some detail scope.
Then I could remove the comparison from the constexpr_string class I guess.